src/cecs277TermProject/RoachMotel.java

```diff
@@ -41,20 +41,21 @@ public RoachMotel(String name, int capacity, File outFile) throws FileNotFoundEx
 	// TODO: [Gust] Add a parameter for list of amenities
 	public void checkIn(RoachColony roachColony) {
 		// TODO: [Tyler] Implement logging this event here
+		// Logging should be mostly finished; just need room methods to be finished - Tyler
 		// TODO: [Ly] Implement checkIn validation
 		System.out.printf("Checking in: %s\n", roachColony);
 		Room room = availableRooms.poll();
 
 		// TODO: [Gust] Implement decorator pattern here for amenities before associating the colony with the room
 		occupiedRooms.put(roachColony, room);
+		logCheckIn(roachColony, this.logger);
```

Comment:
logging should be at the beginning of a method (to keep track of of what events happens). If it's at the end, there maybe exception, and the event would be never logged. You can put a log here as well to log the successful registration event. I don't think we need to create a separate method `logCheckIn()` as log statements should be 1-line statements.

Reply:
I think we need separate methods for checkIn and checkOut because when they check out the system needs to log how they paid, but when checking in they dont.